node/remote/node.go

```diff
@@ -111,20 +111,23 @@ func (cp *Node) Validators(height int64) (*tmctypes.ResultValidators, error) {
 		BlockHeight: height,
 	}
 
-	page := 1
-	perPage := 100
+	maxPages := 2 // maximum number of pages to query
+	currentPage := 1
+	perPage := 100 // maximum 100 entries per page
 	stop := false
 	for !stop {
-		result, err := cp.client.Validators(cp.ctx, &height, &page, &perPage)
-		if err != nil {
-			return nil, err
+		if currentPage <= maxPages {
+			result, err := cp.client.Validators(cp.ctx, &height, &currentPage, &perPage)
+			if err != nil {
+				return nil, err
+			}
+			vals.Validators = append(vals.Validators, result.Validators...)
+			vals.Count += result.Count
+			vals.Total = result.Total
+			currentPage += 1
 		}
-		vals.Validators = append(vals.Validators, result.Validators...)
-		vals.Count += result.Count
-		vals.Total = result.Total
+		stop = vals.Count == vals.Total
```

Comment:
I think we can omit the maxPages variable so if there're > 200 validators it'll still be able to handle it: 
```go
	currentPage := 1
	perPage := 100 // maximum 100 entries per page
	stop := false
	for !stop {
		result, err := cp.client.Validators(cp.ctx, &height, &currentPage, &perPage)
		if err != nil {
			return nil, err
		}

		vals.Validators = append(vals.Validators, result.Validators...)
		vals.Count += result.Count
		vals.Total = result.Total

		currentPage += 1
		stop = vals.Count == vals.Total
	}
```

Reply:
Agree with @huichiaotsou, it's better to be more generic and allow `n` pages of validators, not only 2 cause we might exceed even 200 in the future